bumble/crypto.py

```diff
@@ -23,21 +23,27 @@
 # -----------------------------------------------------------------------------
 from __future__ import annotations
 
+import abc
+import functools
 import logging
 import operator
-
 import secrets
-from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
-from cryptography.hazmat.primitives.asymmetric.ec import (
-    generate_private_key,
-    ECDH,
-    EllipticCurvePrivateKey,
-    EllipticCurvePublicNumbers,
-    EllipticCurvePrivateNumbers,
-    SECP256R1,
-)
-from cryptography.hazmat.primitives import cmac
-from typing import Tuple
+
+from bumble import _crypto
+
+try:
+    from cryptography.hazmat.primitives import ciphers
+    from cryptography.hazmat.primitives.ciphers import algorithms
+    from cryptography.hazmat.primitives.ciphers import modes
+    from cryptography.hazmat.primitives.asymmetric import ec
+    from cryptography.hazmat.primitives import cmac
+
+    _use_builtin_backend = False
```

Comment:
I'd like to propose a slightly different way of splitting the implementation. Instead of relying on `if __use_builtin_backend / else`, we could have 3 files: `crypto.py` with the common parts, `crypto_builtin.py` with the new builtin implementation, and `crypto_cryptography.py` with the one going through the `cryptography` module.
Then, at the top of `crypto.py`, you can `try from crypto_cryptography import EccKey, e, aes_mac except ImportError: from crypto_builtin import EccKey, e, aes_mac`. 
This way, the two variants are well isolated.
(for EccKey, you could have an abstract `BaseEccKey` in `crypto.py` that the two variants would subclass)

Reply:
Sure, this will be much cleaner.

Reply:
Done. Moving these modules into a subpackage _crypto.